src/OrderProcessing/OrderInitialTotalProcessor.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusOrderEditPlugin\OrderProcessing;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusOrderEditPlugin\Entity\InitialTotalAwareOrderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+
+final class OrderInitialTotalProcessor
+{
+    public function __construct(private readonly EntityManagerInterface $entityManager)
+    {
+    }
+
+    public function process(OrderInterface&InitialTotalAwareOrderInterface $order): void
```

Comment:
A lot of times I would introduce a new `OrderInterface` that would inherit from Sylius'. What's your thoughts on this?

Reply:
Agreed 👍